packages/asset-swapper/src/quote_consumers/forwarder_swap_quote_consumer.ts

```diff
@@ -182,59 +182,50 @@ export class ForwarderSwapQuoteConsumer implements SwapQuoteConsumerBase<Forward
             assert.isBigNumber('gasPrice', gasPrice);
         }
 
-        const quoteWithAffiliateFee = affiliateFeeUtils.getSwapQuoteWithAffiliateFee(quote, feePercentage);
-
-        const { orders, feeOrders, worstCaseQuoteInfo } = quoteWithAffiliateFee; // tslint:disable-line:no-unused-variable
+        const { orders, worstCaseQuoteInfo } = quote; // tslint:disable-line:no-unused-variable
 
         // get taker address
         const finalTakerAddress = await swapQuoteConsumerUtils.getTakerAddressOrThrowAsync(this.provider, opts);
-        // if no ethAmount is provided, default to the worst totalTakerTokenAmount
-        const value = ethAmount || worstCaseQuoteInfo.totalTakerTokenAmount;
+        // if no ethAmount is provided, default to the worst totalTakerAssetAmount
+        const ethAmountWithFees = affiliateFeeUtils
+            .getTotalEthAmountWithAffiliateFee(worstCaseQuoteInfo, feePercentage)
+            .plus(worstCaseQuoteInfo.protocolFeeInEthAmount);
```

Comment:
ditto

Reply:
same case, do not need to add protocolFee